updated the install.md for ILIAS 10 (#11194)
ILIAS-eLearning/ILIAS: docs/configuration/install.md

apt install apache2 libapache2-mod-php php php-gd php-xsl php-imagick php-curl php-mysql php-xmlrpc php-soap php-ldap composer
```

Create a directory for the html sources (e.g. `/var/www/ilias`) which is referenced in the apache2 vhost (the apache2 vhost will use `/var/www/ilias/public`) and also a directory outside the web servers docroot (e.g. `/var/www/files`) for files stored by ILIAS. 

> Make sure that the web server user is the owner of the files and directories that were created by changing the group and owner to www-data (on Debian/Ubuntu) or apache (on RHEL).

Reviewer: ```suggestion
Make sure that the web server user is the owner of the files and directories that were created by changing the group and owner to `www-data` (on Debian/Ubuntu), `apache` (on RHEL) or `wwwrun` (on SLES).
```

In addition to the file folder, ILIAS also needs a place to create the log files
(e.g. `/var/www/logs`). The 'ilias.log' can be viewed there later, as well as all
error_log files, which are created in case of errors and are referenced in ILIAS by
an error code.

> Also, to store the ILIAS configuration, which is later used to configure ILIAS, we create the folder /var/www/config. To prevent future issues with npm we create /var/www/.npm which belongs to the web server user.

Reviewer: ```suggestion
Also, to store the ILIAS configuration, which is later used to configure ILIAS, we create the folder `/var/www/config`. To prevent future issues with npm we create `/var/www/.npm` which belongs to the web server user.
```

Author: implemented

max_input_vars = 10000
```

> Restart the apache2 webserver after you installed and configured your dependencies.

Reviewer: ```suggestion
Restart the Apache webserver after you installed and configured your dependencies.
```

Author: decided to stay with apache2

We recommend to clone from GitHub and use git to update the code, since this simplifies
the update to future releases and versions.

Clone the code to the web servers docroot (e.g. `/var/www/ilias`) with the following
commands:

```shell
cd /var/www

> sudo -u www-data git clone --single-branch -b release_10 https://github.com/ILIAS-eLearning/ILIAS.git ilias

Reviewer: Ich halte nicht viel von der Empfehlung für `--single-branch`— das macht es eeeecht fummelig, den Branch in der Zukunft mal zu wechseln.

Author: removed. thanks!

file (located outside your doc-root!) as a parameter:

```shell
sudo -uwww-data php cli/setup.php install /var/www/config/ilias.json
sudo -u www-data php cli/setup.php install /var/www/config/ilias.json
```

The installation will display what currently happens and might prompt you with
questions. You might want to have a look into the [documentation of the command line setup](../../components/ILIAS/setup_/README.md)
or into the help of the program itself `php cli/setup.php help`. It is the tool
to manage and monitor your ILIAS installation.

If you are installing from Git, it is possible that ILIAS will already require a few migrations to the initial
database. Run the setup migration and follow the steps shown. This is also necessary whenever you update your code.

```shell
php cli/setup.php migrate
```

> 

Reviewer 2: Why was this part removed? It's not very likely, but it has happened before. The default data in the template isn't always updated to include all new migrations.

Author: Hi @fwolf-ilias , I never noticed that during a new install of ILIAS. I removed it since I thought it would tidy up the documentation if the migrations only occur in one place of the documentation (before the update). My intention was to make sure that all migrations are done before a major upgrade. I am happy to re-add it if you consider it necessary. Just give me a short hint please.

Reviewer 2: Mhm I see your point. Also it is a bit strange to ask for a migration right on start for the unlikely possibility. Leave the text out.

system in case of failure.
* Internal data within your webroot `<ILIAS-root-folder>/public/data` in our case `/var/www/ilias/public/data`.
* External data, configured in `ilias.json` within `filesystem.data_dir`, in our case `/var/www/files/ilias`.

> * The database, which can easily be dumped by running `mysqldump'.

Reviewer: ```suggestion
* The database, which can easily be dumped by running `mysqldump`.
```

Author: implemented

and `/data` folder to its new destination. Both are now located in the newly created `public` folder. If this `public` folder does not exist yet you 
will have to execute the npm and composer command first: 

```shell
cd /var/www/ilias
sudo -u www-data npm clean-install --omit=dev --ignore-scripts
sudo -u www-data composer install --no-dev
sudo -uwww-data mkdir -p public/Customizing/global/plugins
mv data/* public/data/
mv Customizing/global/plugins/* public/Customizing/global/plugins/
```

Then update the code of your plugins according to their documentation to ensure they are compatible with the new ILIAS version.
If you are **not** using the tar.gz archive to upgrade your release, update your javascript and php dependencies. If you
are using a tar.gz archive and are using plugins, reload your php classmap with `composer du`.

Complete the update of the base system by [updating the database](#update-the-database).

As a last step, you should log in with a User using your custom skin. If everything
works fine, change back from Delos to your custom system style. If not, you probably
will need to update your style to match the new release.

<a name="update-the-database"></a>

> ## Update the Database and doing migrations to IRSS

Reviewer: ```suggestion
## Update the Database and run the Migrations
```

Reviewer: Können auch andere Migrations sein?

Author: removed "to IRSS".

some time, but which can also be executed while the installation is in productive use.

Run the `status` command on the command line to check if there are any updates
available and if ILIAS is responding. After this you need to perform the update. 

> Please make sure to check for migrations before you run the update. If there are migrations left

Reviewer 2: In my opinion, this is completely wrong. The structure of the database must correspond to the expectations of the code at all times. This also applies to the migration code. If the migrations are executed beforehand and access the database, what structure does the migration step expect? Without an update, there are endless possibilities; after the update, there is only one (that of the current status defined by the update steps).

Author: As I said earlier my intention was to make sure that all migrations are run before a major upgrade. If I understand you correctly the proposal would be to run all migrations after an update but also before an update? What occured to me from time to time was that migrations are increasing without doing an update.

Reviewer 2: Ah I understand. In case of a Major release, this is a hint we have to add. Thanks for that. This hint is should be at the top of the chapter "#Major Upgrade", in "#Update the Database" it is too late for that, as the code update is already done.

Author: I added this at the beginning of the major upgrade section.

> please make sure that these are done before updating.

Reviewer: ```suggestion
please make sure to run these before updating.
```

Author: implemented

or into the help of the program itself `php cli/setup.php help`. It is the tool
to manage and monitor your ILIAS installation.

As soon as all migrations are done you can do the database update:

```shell
cd /var/www/ilias
sudo -u www-data php cli/setup.php update
```

Database updates are performed in steps; it might happen that a step fails, e.g. due
to some edge case or inconsistency in existing data, files, etc.

> In this case, a consecutive command `php setup/setup.php update` will error with

Reviewer: ```suggestion
In this case, a consecutive command `php setup/setup.php update` will fail with
```

Author: implemented

```shell
php cli/setup.php achieve database.resetFailedSteps
```
However, be sure to understand the cause for the failing steps and take care of it before
resetting and running update again.

<a name="information-updates"></a>
## Information on Updates

To keep your ILIAS Installation secure and healthy it is important that you keep
it up to date. To get information about updates and security fixes you should
consider subscribing to the [ILIAS Admin Mailing-List](https://lists.ilias.de/postorius/lists/ilias-admins.lists.ilias.de/). Information on the 
new versions, such as Important Changes, Known Issues, Changed Behaviour and Fixed 

> Issues, can be found in the release notes in [Download & Releases](https://docu.ilias.de/go/pg/197850_35).

Reviewer 2: I don't want to change that link after every update. The old link references the  complete Download & Releases object, the new link a sub page of this object which is outdated rather quickly.

Author: As soon as ILIAS 11.0 stable arrives the link will show to this, right? I thought it would be good if each link points to the first stable release of each version. On the other hand I can well understand that there's a need for a permanent link. I'll leave this decision up to you. I'm fine with either way.

Reviewer 2: The permanent link is I used links to the whole learning module. If you never visited this LM or as anonymous, it opens the first page of the first chapter a.k.a. the highest stable release. If you are logged it and visited it before, it will show the last visited page. The permanent links with 'pg' are linking a specified page. It will stay at 10.0 as long as you don't change the link. 

So I opt to keep the old link https://docu.ilias.de/go/lm/35.

Author: implemented the old link again.

> Issues, can be found in the release notes in [Download & Releases](https://docu.ilias.de/go/pg/197850_35).

Reviewer 2: ```suggestion
Issues, can be found in the release notes in [Download & Releases](https://docu.ilias.de/go/lm/35).
```
